rent.go

```diff
@@ -0,0 +1,81 @@
+package iotago
+
+import (
+	"errors"
+	"fmt"
+)
+
+// VByteCostFactor defines the type of the virtual byte cost factor.
+type VByteCostFactor uint64
+
+const (
+	// VByteCostFactorData defines the multiplier for data fields.
+	VByteCostFactorData VByteCostFactor = 1
+	// VByteCostFactorKey defines the multiplier for fields which can act as keys for lookups.
+	VByteCostFactorKey VByteCostFactor = 10
+)
+
+var (
+	// ErrVByteRentNotCovered gets returned when a NonEphemeralObject does not cover the state rent
+	// cost which are calculated from its virtual byte costs.
+	ErrVByteRentNotCovered = errors.New("virtual byte rent costs not covered")
+
+	// ZeroRentParas are test parameters for de/serialization using zero vbyte rent cost.
+	// Only use this var in testing. Do not modify.
+	ZeroRentParas = &DeSerializationParameters{RentStructure: &RentStructure{
+		VByteCost:    0,
+		VBFactorData: 0,
+		VBFactorKey:  0,
+	}}
+)
+
+// Multiply multiplies in with this factor.
+func (factor VByteCostFactor) Multiply(in uint64) uint64 {
+	return uint64(factor) * in
+}
+
+// With joins two factors with each other.
+func (factor VByteCostFactor) With(other VByteCostFactor) VByteCostFactor {
+	return factor + other
+}
+
+// RentStructure defines the parameters of rent cost calculations on objects which take node resources.
+type RentStructure struct {
+	// Defines the rent of a single virtual byte denoted in IOTA tokens.
+	VByteCost uint64 `json:"vByteCost"`
+	// Defines the factor to be used for data only fields.
+	VBFactorData VByteCostFactor `json:"vByteFactorData"`
+	// defines the factor to be used for key/lookup generating fields.
+	VBFactorKey VByteCostFactor `json:"vByteFactorKey"`
+}
+
+// CoversStateRent tells whether given this NonEphemeralObject, the given rent fulfils the renting costs
+// by examining the virtual bytes cost of the object.
+// Returns the minimum rent computed and an error if it is not covered by rent.
+func (vbcs *RentStructure) CoversStateRent(object NonEphemeralObject, rent uint64) (uint64, error) {
+	minRent := vbcs.VByteCost * object.VByteCost(vbcs, nil)
+	if rent < minRent {
+		return 0, fmt.Errorf("%w: needed %d but only got %d", ErrVByteRentNotCovered, minRent, rent)
+	}
+	return minRent, nil
+}
+
+// MinStorageDeposit returns the minimum renting costs for an BasicOutput which returns
+// a StorageDepositReturnUnlockCondition amount back to the origin sender.
+func (vbcs *RentStructure) MinStorageDeposit(sender Address) uint64 {
```

Comment:
Rename `MinStorageDeposit` to `ReturnOutputMinStorageDeposit` maybe? :)
Just cosmetics ofc.

Reply:
e9e9364b598165fbb24366858ffcbc4eae91a134